stable/prometheus/templates/alertmanager-pvc.yaml

```diff
@@ -22,6 +17,13 @@ metadata:
 spec:
   accessModes:
 {{ toYaml .Values.alertmanager.persistentVolume.accessModes | indent 4 }}
+{{- if .Values.alertmanager.persistentVolume.storageClass }}
+{{- if (eq "-" .Values.alertmanager.persistentVolume.storageClass) }}
+  storageClassName: ""
+{{- else }}
+  storageClassName: "{{ .Values.alertmanager.persistentVolume.storageClass }}"
+{{- end }}
+{{- end }}
```

Comment:
This seems like a reasonable approach. Provides the necessary options for 1.6+, and users can still use on <1.6 by leaving `storageClass` unset and providing the annotation.

Ping @unguiculus @viglesiasce @lachie83 @prydonius @seanknox @linki @sameersbn @foxish: is this a/the pattern we want to adopt repo-wide?